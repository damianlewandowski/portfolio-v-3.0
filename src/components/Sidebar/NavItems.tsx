import React from "react";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import List from "@material-ui/core/List";
import PersonIcon from "@material-ui/icons/Person";
import UpdateIcon from "@material-ui/icons/Update";
import ListAltIcon from "@material-ui/icons/ListAlt";
import { green, purple } from "@material-ui/core/colors";
import EqualizerIcon from "@material-ui/icons/Equalizer";
import { IconItem } from "../interfaces";

const NavItems = () => {
  const items: IconItem[] = [
    { text: "Profile", icon: <PersonIcon color={"primary"} /> },
    { text: "Experience", icon: <UpdateIcon color={"secondary"} /> },
    {
      text: "Projects",
      icon: <ListAltIcon style={{ color: purple[500] }} />
    },
    {
      text: "Skills",
      icon: <EqualizerIcon style={{ color: green[500] }} />
    }
  ];
  return (
    <List>
      {items.map(item => (
        <ListItem button key={item.text}>
          <ListItemIcon>{item.icon}</ListItemIcon>
          <ListItemText primary={item.text} />
        </ListItem>
      ))}
    </List>
  );
};

export default NavItems;
