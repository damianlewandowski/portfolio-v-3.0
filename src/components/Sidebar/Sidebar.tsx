import React from "react";
import { createStyles, Theme, makeStyles } from "@material-ui/core/styles";
import Drawer from "@material-ui/core/Drawer";
import Typography from "@material-ui/core/Typography";
import { Box } from "@material-ui/core";
import NavItems from "./NavItems";

const drawerWidth = 250;

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    drawer: {
      width: drawerWidth,
      flexShrink: 0
    },
    drawerPaper: {
      width: drawerWidth
    },
    sideNav: {
      height: "100vh"
    },
    box: {
      position: "relative",
      width: "100%",
      "&::after": {
        content: "''",
        display: "block",
        paddingBottom: "100%"
      }
    },
    textContainer: {
      position: "absolute",
      left: "10%",
      top: "50%",
      textTransform: "capitalize"
    }
  })
);

export default function PermanentDrawerLeft() {
  const classes = useStyles();

  return (
    <Drawer
      className={classes.drawer}
      variant="permanent"
      classes={{
        paper: classes.drawerPaper
      }}
      anchor="left"
    >
      <Box
        className={classes.box}
        bgcolor="primary.main"
        color="primary.contrastText"
      >
        <Typography className={classes.textContainer} variant="h5">
          damian
          <br /> <strong>lewandowski</strong>
        </Typography>
      </Box>

      <NavItems />
    </Drawer>
  );
}
