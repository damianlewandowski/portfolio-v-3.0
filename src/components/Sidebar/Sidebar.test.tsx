import React from "react";

import Sidebar from "./Sidebar";
import {mount, shallow} from "enzyme";
import {List} from "@material-ui/core";

it("renders correctly", () => {
  const wrapper = shallow(<Sidebar />);

  expect(wrapper).toMatchSnapshot();
});

// it("has correct button label", () => {
//   const wrapper = mount(
//     <ButtonWithTextInput btnLabel={"Sie buton"} textInputValue={"Zurich"} />
//   );
//   const text = wrapper.find("CustomButton").text();
//   expect(text).toEqual("Sie buton");
// });

// it("has correct text input value", () => {
//   const wrapper = mount(
//     <ButtonWithTextInput btnLabel={"Sie buton"} textInputValue={"Zurich"} />
//   );
//   const text = wrapper.find("CustomTextInput").props().textInputValue;
//   expect(text).toEqual("Zurich");
// });

// it("test visual result after click - call setNewCityName", () => {
//   const wrapper = mount(
//     <ButtonWithTextInput btnLabel={"Sie buton"} textInputValue={"Zurich"} />
//   );

//   wrapper
//     .find("button")
//     .first()
//     .simulate("click");

//   expect(wrapper.find("CustomTextInput").props().textInputValue).toEqual(
//     "Santa Maria"
//   );
// });
