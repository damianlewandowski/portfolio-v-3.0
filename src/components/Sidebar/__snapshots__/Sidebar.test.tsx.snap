// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<WithStyles(ForwardRef(Drawer))
  anchor="left"
  className="makeStyles-drawer-1"
  classes={
    Object {
      "paper": "makeStyles-drawerPaper-2",
    }
  }
  variant="permanent"
>
  <Styled(MuiBox)
    bgcolor="primary.main"
    className="makeStyles-box-4"
    color="primary.contrastText"
  >
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-textContainer-5"
      variant="h5"
    >
      damian
      <br />
       
      <strong>
        lewandowski
      </strong>
    </WithStyles(ForwardRef(Typography))>
  </Styled(MuiBox)>
  <NavItems />
</WithStyles(ForwardRef(Drawer))>
`;
